### PR TITLE
start Migration to typescript

### PR DESCRIPTION
**Pull request by JcdeA**
Typescript will help prevent future bugs, and will be useful anyways while developing.
This PR will also dynamically load components for the grapesjs editor, which improves speed vastly. 

The contentprovider component used to be over 200kbs, but now is only 3.4kbs.

### CODE REVIEW

**Comment by JcdeA** [comment]
Shoot I forgot that I removed the part that injects tailwind from a cdn in the library..

In this pr it removes those and the user has to install tailwind themselves.

Actually, it might be  a better Idea to give the user a choice.. use tailwind from cdn if they're lazy, or install tailwind themselves for better performance.

There isn't any way to trigger postcss(in nextjs) to run without creating a postcss config file in the users nextjs app as far as I know.  (tailwind is a postcss plugin)

**Comment by JcdeA** [comment]
image
Right is before, Left is after this pr.
Massive improvement because it doesn't load all the components for the grapejs editor in production mode.

**Comment by JcdeA** [comment]
Hmm, this doesn't seem to work with blitz.js very well..

Blitzjs's webpack isn't recognizing the grapes. Min-(hash).js as. Js file for some reason

**Comment by LiveDuo** [comment]
> Hmm, this doesn't seem to work with blitz.js very well..
> 
> Blitzjs's webpack isn't recognizing the grapes. Min-(hash).js as. Js file for some reason

That's strange. I thought blitzjs was built on Next.js.

**Comment by LiveDuo** [comment]
> image
> Right is before, Left is after this pr.
> Massive improvement because it doesn't load all the components for the grapejs editor in production mode.

Forgot to say. That improvement is huge! It makes attractive to SEO people and better position when the blogging system is ready!

**Comment by JcdeA** [comment]
Yep! It used to load ALL of the tailblocks components even in production. We were just loading 250+kbs of unused code.

I just made it dynamically load grapes.js using es2020 dynamic imports.

File: lib/destack/src/components/initEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { loadTraits } from '../lib/traits'
+import { loadComponents } from '../lib/components'
+import { loadBlocks } from '../lib/blocks'
+import { fetchJSON, escapeName } from '../utils'
+import { appendCss } from '../lib/css'
+import { handleEvents } from '../lib/events'
+
+const initEditor = async ({ server = true }): Promise<void> => {
+  const grapesjs = await import('grapesjs')
+  /*       const uploadFile = (e): void => {
```
Comment (LiveDuo) [inline lib/destack/src/components/initEditor.tsx:11]: I notice that `uploadFile` is broken (before this PR). 

Will fix it later and check these changes in depth later when I get to fix it.

**Review by LiveDuo — CHANGES_REQUESTED** [review]
I think we should add`.tsbuildinfo` (and `lib/destack/.tsbuildinfo`) and `.vscode` to `.gitignore`. Some like to have VSCode config commited into their repo to have a consistent work environment but since it's an open source project it might be better to let contributors choose their own config.

**Comment by LiveDuo** [comment]
I was thinking about TailwindCSS integration and what's the best way to go forward with it. For the time being it might be better to seperate Tailwind changes into a seperate PR.

There are a few things to consider further before moving on with that change. On the other hand Typescript support on the other hand seems good to go. Same with dynamically loading Grapes.js!

**Comment by JcdeA** [comment]
> I think we should add`.tsbuildinfo` (and `lib/destack/.tsbuildinfo`) and `.vscode` to `.gitignore`. Some like to have VSCode config commited into the repo to have a consistent work environment but since it's an open source project it might be better to let contributors choose their own config.

Thanks!

**Comment by JcdeA** [comment]
see #18

**Pull request closed without merging**